### PR TITLE
Pass headers to s3-sync to be able to set metadata in objects

### PR DESCRIPTION
**Pull request by jrbenito**
s3-sync has the ability to receive headers as arguments and pass them to
AWS S3 as metadata headers. This modification proposes to set headers at
configuration so they are written in the objects when deploying.

Main motivation is to be able to pass CacheControl header and have
Cache-Control set for objects ovoiding "Leverage Browser Caching"
throwed by page speed analyzers.

Attention: Metadata are written equal to all objects (files) stored to
S3 and there is no way to tailor it by file type at this moment.

Signed-off-by: Josenivaldo Benito Jr jrbenito@benito.qsl.br

### CODE REVIEW

**Comment by nt3rp** [comment]
Sorry for the delay; I wasn't receiving notifications.

I think this should be possible with the [`s3`](https://www.npmjs.com/package/s3) library (which is now used in this project). If you can fix the merge conflicts, I would happily merge in your request :)

**Comment by jrbenito** [comment]
@nt3rp ,

Hi, I will check S3 library and update PR.

I have switched to sync-s3-aws in my repo, I don ´t know diferences but I liked sync dir  feature from S3 you pointed out.

Regards,
Benito

**Comment by jrbenito** [comment]
Hi @nt3rp,

I´ve updated PR.

File: lib/deployer.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ module.exports = function(args) {
   var publicDir = this.config.public_dir;
   var log = this.log;
 
+  var customHeaders = args.hearders || {};
```
Comment (nt3rp) [inline lib/deployer.js:19]: Shouldn't this be `arg.headers`?

**Comment by jrbenito** [comment]
@nt3rp , 

Thanks for point this. I will pull of this PR for while, seems I tested against wrong branch in my Git repository, so everything I put here was not tested properly. **My bad!**

As soon as I test it right I correct PR and send it again. Hold on this one.

**Pull request closed without merging**